Resources/public/js/views/services/plugins/ez-locationsearchplugin.js

```diff
@@ -222,31 +223,41 @@ YUI.add('ez-locationsearchplugin', function (Y) {
          * @param {Array|Null} locationStructArr
          * @param {Function} callback
          */
-        _loadContentForLocations: function (locationStructArr, callback) {
-            var tasks = new Y.Parallel(),
-                loadContentError = false;
+        _loadContentForLocations: function (viewName, locationStructArr, callback) {
+            var contentService = this.get('host').get('capi').getContentService(),
+                contentIdCriterions = '',
+                loadContentError = false,
+                that = this,
+                query;
 
             Y.Array.each(locationStructArr, function (locationStruct, i) {
-                var content = new Y.eZ.Content(
-                        {id: locationStruct.location.get('contentInfo').get('id')}
-                    ),
-                    capi = this.get('host').get('capi'),
-                    options = {api: capi},
-                    endContentLoad = tasks.add(function (error, response) {
-                        if (error) {
-                            loadContentError = true;
-                            return;
-                        }
-
-                        locationStructArr[i].content = content;
-                    });
-
-                content.load(options, endContentLoad);
+                contentIdCriterions =
```

Comment:
you should have a look at http://yuilibrary.com/yui/docs/api/classes/Array.html#method_reduce which would be bit more elegant